tests/test_tempcompare.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from etl.tempcompare import df_equals
+
+
+def test_df_equals():
+    # Test with two identical dataframes
+    df1 = pd.DataFrame({"col1": [1, 2, 3], "col2": ["a", "b", "c"]})
+    df2 = pd.DataFrame({"col1": [1, 2, 3], "col2": ["a", "b", "c"]})
+    assert df_equals(df1, df2).all().all()
+
+    # Test with two different dataframes
+    df1 = pd.DataFrame({"col1": [1, 2, 3], "col2": ["a", "b", "c"]})
+    df2 = pd.DataFrame({"col1": [1, 2, 3], "col2": ["d", "e", "f"]})
+    assert not df_equals(df1, df2).all().all()
+
+    # Test with numeric differences within tolerance
+    df1 = pd.DataFrame({"col1": [1.0, 2.0, 3.0]})
+    df2 = pd.DataFrame({"col1": [1.0, 2.00001, 3.0]})
+    assert df_equals(df1, df2, absolute_tolerance=1e-05).all().all()
+
+    # Test with categorical columns
+    df1 = pd.DataFrame({"col1": pd.Categorical(["a", "b", "c"])})
+    df2 = pd.DataFrame({"col1": pd.Categorical(["a", "b", "c"])})
+    assert df_equals(df1, df2).all().all()
+
+
+def test_df_equals_differing():
+    # Test with numeric differences exceeding absolute tolerance
+    df1 = pd.DataFrame({"col1": [1.0, 2.0, 3.0]})
+    df2 = pd.DataFrame({"col1": [1.0, 2.0001, 3.0]})
+    assert not df_equals(df1, df2, absolute_tolerance=1e-05).all().all()
+
+    # Test with categorical columns that are not identical
+    df1 = pd.DataFrame({"col1": pd.Categorical(["a", "b", "c"])})
+    df2 = pd.DataFrame({"col1": pd.Categorical(["a", "b", "d"])})
+    assert not df_equals(df1, df2).all().all()
+
+    # Test with one column having NaN values
+    df1 = pd.DataFrame({"col1": [1, 2, 3], "col2": ["a", "b", "c"]})
+    df2 = pd.DataFrame({"col1": [1, 2, np.nan], "col2": ["a", "b", "c"]})
```

Comment:
What about testing whether two dataframes having nan on the same position are considered equal?
```
df1 = pd.DataFrame({"col1": [1, 2, np.nan], "col2": ["a", "b", "c"]})
df2 = pd.DataFrame({"col1": [1, 2, np.nan], "col2": ["a", "b", "c"]})
assert df_equals(df1, df2).all().all()
```

Reply:
Added that. Shame on you chatgpt...